src/handle/why3_tactic.ml

```diff
@@ -42,30 +42,84 @@ let why3_env : Why3.Env.env =
 
 (** Builtin configuration for propositional logic. *)
 type config =
-  { symb_P   : sym (** Encoding of propositions. *)
-  ; symb_T   : sym (** Encoding of types.        *)
-  ; symb_or  : sym (** Disjunction(∨) symbol.    *)
-  ; symb_and : sym (** Conjunction(∧) symbol.    *)
-  ; symb_imp : sym (** Implication(⇒) symbol.    *)
-  ; symb_bot : sym (** Bot(⊥) symbol.            *)
-  ; symb_top : sym (** Top(⊤) symbol.            *)
-  ; symb_not : sym (** Not(¬) symbol.            *) }
+  { symb_P     : sym (** Encoding of propositions. *)
+  ; symb_T     : sym (** Encoding of types.        *)
+  ; symb_or    : sym (** Disjunction(∨) symbol.    *)
+  ; symb_and   : sym (** Conjunction(∧) symbol.    *)
+  ; symb_imp   : sym (** Implication(⇒) symbol.    *)
+  ; symb_bot   : sym (** Bot(⊥) symbol.            *)
+  ; symb_top   : sym (** Top(⊤) symbol.            *)
+  ; symb_not   : sym (** Not(¬) symbol.            *)
+  ; symb_ex    : sym (** Exists(∃) symbol.         *)
+  ; symb_all   : sym (** Forall(∀) symbol.         *)
+  ; symb_codes : sym (** Type of encoded types.    *) }
 
 (** [get_config ss pos] build the configuration using [ss]. *)
 let get_config : Sig_state.t -> Pos.popt -> config = fun ss pos ->
   let builtin = Builtin.get ss pos in
-  { symb_P   = builtin "P"
-  ; symb_T   = builtin "T"
-  ; symb_or  = builtin "or"
-  ; symb_and = builtin "and"
-  ; symb_imp = builtin "imp"
-  ; symb_bot = builtin "bot"
-  ; symb_top = builtin "top"
-  ; symb_not = builtin "not" }
+  { symb_P     = builtin "P"
+  ; symb_T     = builtin "T"
+  ; symb_or    = builtin "or"
+  ; symb_and   = builtin "and"
+  ; symb_imp   = builtin "imp"
+  ; symb_bot   = builtin "bot"
+  ; symb_top   = builtin "top"
+  ; symb_not   = builtin "not"
+  ; symb_ex    = builtin "ex"
+  ; symb_all   = builtin "all"
+  ; symb_codes = builtin "Set" }
 
 (** A map from lambdapi terms to Why3 constants. *)
 type cnst_table = (term * Why3.Term.lsymbol) list
```

Comment:
We could use a more efficient data structure like Map using Term.cmp.

Reply:
For this we would have to ensure that `forall t, TMap.find t m = List.assoc_eq (Eval.eq_modulo []) t l`, and I don't think we need to be more efficient here

Reply:
Right.